### ISSUE TITLE
Tiled communication issue with read_data when using shrinkwrap boundary conditions

### ISSUE THREAD

**Issue by akohlmey** [lammps/lammps#1249]
## Summary

With the provided input deck, the `read_data` command fails when switching to tiled communication *beforehand*. Moving the `comm_style` command *after* reading the data file results in correct execution.

## Type of Issue

Bug report

## LAMMPS Version (Bug Report)

Last checked with LAMMPS version 27 November 2018.

## Expected Behavior (Bug Report)

LAMMPS should be able to read data files with either 'brick' or 'tiled' communication and with all available boundary types. In this case the combination of 'tiled' with 'shrinkwrap' (either 's' or 'm') fails. Using other boundary conditions seem to work.

## Actual Behavior (Bug Report)

LAMMPS fails with this message from the MPI library:
```
LAMMPS (27 Nov 2018)
OMP_NUM_THREADS environment is not set. Defaulting to 1 thread.
(src/comm.cpp:87)
  using 1 OpenMP thread(s) per MPI task
Reading data file ...
  orthogonal box = (1e-05 1e-05 1e-05) to (109 109 27)
  1 by 1 by 1 MPI processor grid
  reading atoms ...
  16000 atoms
Fatal error in MPI_Irecv: Invalid argument, error stack:
MPI_Irecv(153): MPI_Irecv(buf=0x5b640b0, count=1, MPI_INT, src=0,
tag=0, MPI_COMM_WORLD, request=(nil)) failed
MPI_Irecv(100): Null pointer in parameter request
```
## Steps to Reproduce (Bug Report)

Input deck to reproduce is here: 
[tiled-comm-bug.tar.gz](https://github.com/lammps/lammps/files/2658149/tiled-comm-bug.tar.gz)

## Further Information, Files, and Links

Issue was reported by Adrian Diaz on lammps-users mailing list.

**Comment by Adrian-Diaz** [comment]
Yea the segfault arises due to the point drop function pointer in comm_tiled.cpp being called in exchange before the function pointer is initialized when using shrink wrap. Fixing that entirely however may not be too pretty since it requires moving, in addition to the function pointer initialization, the exchange setup towards the end of the setup routine somewhere that its called before exchange in read_data.

**Comment by akohlmey** [comment]
For the time being, it should be sufficient to simply add a check for shrinkwarp+tiled and error out with a message to switch to tiled communication **after** read_data. I currently cannot imagine a setup where we would absolutely need tiled communication to work this early.

**Comment by akohlmey** [comment]
fixed by PR #1325
